src/server/document-controller.js

```diff
@@ -192,7 +192,12 @@ class DocumentController {
 
     router.get(
       '/docs/:docKey*',
-      [validateParams(getDocumentParamsSchema), validateQuery(getDocumentQuerySchema)],
+      validateParams(getDocumentParamsSchema),
+      validateQuery(getDocumentQuerySchema),
+      needsPermission({
+        value: permissions.EDIT_DOC,
+        condition: req => [DOC_VIEW_QUERY_PARAM.edit, DOC_VIEW_QUERY_PARAM.history].includes(req.query.view)
```

Comment:
Here I find `Object.keys(DOC_VIEW_QUERY_PARAM).includes(req.query.view)` would be better

Reply:
True, me too. Changed!